server/src/main/java/org/apache/iotdb/db/monitor/StatMonitor.java

```diff
@@ -16,317 +16,291 @@
     }
   }
 
-  public void recovery() {
-    // // restore the FildeNode Manager TOTAL_POINTS statistics info
+  private void registSeriesToMManager(PartialPath fullPath) throws MetadataException {
+    if (!mManager.isPathExist(fullPath)) {
+      mManager.createTimeseries(fullPath,
+          TSDataType.valueOf(MonitorConstants.INT64),
+          TSEncoding.valueOf("RLE"),
+          TSFileDescriptor.getInstance().getConfig().getCompressor(),
+          Collections.emptyMap());
+    }
   }
 
-  void activate() {
-    service = IoTDBThreadPoolFactory.newScheduledThreadPool(1,
-        ThreadName.STAT_MONITOR.getName());
-    service.scheduleAtFixedRate(
-        new StatBackLoop(), 1, backLoopPeriod, TimeUnit.SECONDS);
+  public void updateStatStorageGroupValue(String storageGroupName, int successPointsNum) {
+    List<PartialPath> monitorSeries = monitorSeriesMap.get(storageGroupName);
+    // update TOTAL_POINTS of each storage group
+    cachedValueMap.computeIfPresent(monitorSeries.get(0),
+        (key, oldValue) -> oldValue + successPointsNum);
   }
 
-  void clearIStatisticMap() {
-    statisticMap.clear();
+  public void updateStatGlobalValue(int successPointsNum) {
+    List<PartialPath> monitorSeries = monitorSeriesMap
+        .get(MonitorConstants.STAT_STORAGE_GROUP_NAME);
+    for (int i = 0; i < monitorSeries.size() - 1; i++) {
+      // 0 -> TOTAL_POINTS, 1 -> REQ_SUCCESS, 2 -> REQ_FAIL
+      switch (i) {
+        case 0:
+          cachedValueMap.computeIfPresent(monitorSeries.get(i),
+              (key, oldValue) -> oldValue + successPointsNum);
+          break;
+        case 1:
+          cachedValueMap.computeIfPresent(monitorSeries.get(i),
+              (key, oldValue) -> oldValue + 1);
+          break;
+      }
+    }
   }
 
-  public long getNumBackLoop() {
-    return numBackLoop.get();
+  public void updateFailedStatValue() {
+    PartialPath failedSeries = monitorSeriesMap
+        .get(MonitorConstants.STAT_STORAGE_GROUP_NAME).get(2);
+    cachedValueMap.computeIfPresent(failedSeries, (key, oldValue) -> oldValue + 1);
   }
 
   /**
-   * register class which implemented IStatistic interface into statisticMap
-   *
-   * @param path the stat series prefix path, like root.stat.file.size
-   * @param iStatistic instance of class which implemented IStatistic interface
+   * Generate tsRecords for stat parameters and insert them into StorageEngine.
    */
-  public void registerStatistics(String path, IStatistic iStatistic) {
-    synchronized (statisticMap) {
-      logger.debug("Register {} to StatMonitor for statistics service", path);
-      this.statisticMap.put(path, iStatistic);
+  public void cacheStatValue() {
+    StorageEngine storageEngine = StorageEngine.getInstance();
+    long insertTime = System.currentTimeMillis();
+    for (Entry<PartialPath, Long> cachedValue : cachedValueMap.entrySet()) {
+      TSRecord tsRecord = new TSRecord(insertTime, cachedValue.getKey().getDevice());
```

Comment:
`TSRecord` should be refined, as we do not need to generate a `String` in TsRecord.